dotnet/G3DViewer/MainViewModel.cs

```diff
@@ -250,35 +262,177 @@ internal int AddG3DData(Ara3D.IG3D g3dFile)
             Ara3D.IArray<int> faceSizes = Ara3D.G3DExtensions.FaceSizes(g3dFile);
 
             var vertexData = g3dFile.VertexAttribute.ToVector3s();
-            var indexData = g3dFile.IndexAttribute.ToInts();
+            var uvData = Ara3D.G3DExtensions.UVs(g3dFile);
+            var indexData = Ara3D.G3DExtensions.CornerVertexIndices(g3dFile);
 
+            float aabbMinX = float.MaxValue;
+            float aabbMaxX = float.MinValue;
+            float aabbMinY = float.MaxValue;
+            float aabbMaxY = float.MinValue;
+            float aabbMinZ = float.MaxValue;
+            float aabbMaxZ = float.MinValue;
 
-            var b1 = new MeshBuilder(true, true, true);
-            var l1 = new LineBuilder();
+            Vector3Collection sharpDxVectos = new Vector3Collection(vertexData.Count);
+            Vector2Collection sharpDxUVs = new Vector2Collection(vertexData.Count);
 
+            displayStats.NumVertices += vertexData.Count;
+            displayStats.NumFaces += faceCount;
+
+            for (int vertexIndex = 0; vertexIndex < vertexData.Count; vertexIndex++)
+            {
+                sharpDxVectos.Add(Ara3DToSharpDX(vertexData[vertexIndex]));
+                sharpDxUVs.Add(Ara3DToSharpDX(uvData[vertexIndex]));
+
+                aabbMinX = Math.Min(aabbMinX, vertexData[vertexIndex].X);

```

Comment:
This code for computing a bounding box already exists. I would use `Ara3D.Box.Create()` , but actually there is already an extensions method on `IGeometry` for getting the bounding box.

Reply:
This would require another loop over the data, an I'm not sure if it would be worthwhile...